native/kotlin/api/kotlin/src/main/kotlin/rs/wordpress/api/kotlin/WpRequestExecutor.kt

```diff
@@ -91,4 +116,83 @@ class WpRequestExecutor(
     override suspend fun sleep(millis: ULong) {
         delay(millis.toLong())
     }
+
+    private fun getClient(): OkHttpClient {
+        if (allowedHostnames.isEmpty()) {
+            return okHttpClient
+        }
+
+        val clientCertificates = HandshakeCertificates.Builder()
+            .addPlatformTrustedCertificates()
+            .addInsecureHost(allowedHostnames.first())
+            .build()
+
+        return okHttpClient.newBuilder()
+            .hostnameVerifier(WpRequestExecutorHostnameVerifier(allowedHostnames))
+            .sslSocketFactory(clientCertificates.sslSocketFactory(), clientCertificates.trustManager)
+            .build()
+    }
+
+    @Suppress("UnusedParameter")
+    private fun handleInvalidSSLError(e: SSLPeerUnverifiedException, request: Request) {
+        // It's kind of weird and annoying that we need to make a second request to get
+        // this data, but it doesn't seem like we can get it from the response or the
+        // `SSLPeerUnverifiedException` directly.
```

Comment:
Maybe I am misunderstanding it, but this seems like a major security issue to me. We are retrying the request while allowing insecure connections without any explicit consent from the client. I think this is flow is only valid if it's done by the client after a user prompt. The information about invalid SSL certificate is definitely not important enough to throw away security considerations.

Please let me know if I am misunderstanding or misinterpreting it.

Reply:
We're not actually re-sending the request – we're issuing a request to the same URL. 

The difference there is that no secret information is being transferred (for instance, the `Authorization` header will always be empty).

I could see an argument for stripping query params too, and we could make that change, but I wanted to start with that clarification.